go/libraries/doltcore/sqle/dtablefunctions/dolt_diff_summary.go

```
@@ -438,3 +452,45 @@ func getRowFromSummary(ds *diff.TableDeltaSummary) sql.Row {
 		ds.SchemaChange,           // schema_change
 	}
 }
+
+// getIgnorePatternsFromContext retrieves ignore patterns from the dolt_ignore table.
+func getIgnorePatternsFromContext(ctx *sql.Context, database sql.Database) (doltdb.IgnorePatterns, error) {
+	sess := dsess.DSessFromSess(ctx.Session)
+	dbName := database.Name()
+	roots, ok := sess.GetRoots(ctx, dbName)
+	if !ok {
+		return nil, fmt.Errorf("Could not load database %s", dbName)
+	}
+
+	ignorePatternMap, err := doltdb.GetIgnoredTablePatterns(ctx, roots, []string{""})
+	if err != nil {
+		return nil, err
+	}
+
+	// Return patterns for default schema
+	return ignorePatternMap[""], nil
+}
+
+// shouldIgnoreDelta determines if a table delta should be ignored based on dolt_ignore patterns.
+// This follows the same logic as the dolt diff command: only "added" or "dropped" tables
+// can be ignored, not modified/renamed tables.
+func shouldIgnoreDelta(delta diff.TableDelta, ignorePatterns doltdb.IgnorePatterns) bool {
+	if delta.IsAdd() {
+		ignoreResult, err := ignorePatterns.IsTableNameIgnored(delta.ToName)
+		if err != nil {
+			return false // On error, don't ignore
```

Ignoring the error isn't ideal. This would effectively mean that the regex was garbage to begin with, and the user should probably know that. Both places this method is used can bubble up errors, so I'll add a commit which does that.